browser/ui/browser_commands.cc

```diff
@@ -84,50 +84,35 @@ void OpenGuestProfile() {
   using DistillState = speedreader::SpeedreaderTabHelper::DistillState;
-  speedreader::SpeedreaderService* service =
-      speedreader::SpeedreaderServiceFactory::GetForProfile(browser->profile());
-  if (service) {
-    WebContents* contents = browser->tab_strip_model()->GetActiveWebContents();
-    if (contents) {
-      auto* tab_helper =
-          speedreader::SpeedreaderTabHelper::FromWebContents(contents);
-      if (!tab_helper)
-        return;
-
-      const bool speedreader_enabled = tab_helper->IsSpeedreaderEnabled();
-      const DistillState state = tab_helper->PageDistillState();
-
-      if (state == DistillState::kSpeedreaderMode) {
-        // The page was distilled by Speedreader and the user clicked the icon.
-        // Show per-domain blacklists.
+  WebContents* contents = browser->tab_strip_model()->GetActiveWebContents();
+  if (contents) {
+    auto* tab_helper =
+        speedreader::SpeedreaderTabHelper::FromWebContents(contents);
+    if (!tab_helper)
+      return;
+
+    const DistillState state = tab_helper->PageDistillState();
+    switch (state) {
+      case DistillState::kSpeedreaderMode:
+      case DistillState::kSpeedreaderOnDisabledPage:
         tab_helper->ShowSpeedreaderBubble();
-      } else if (state == DistillState::kReaderMode) {
-        // The page was distilled manually and the user clicked the icon.
-        // Three things can happen:
-        //   (1) Speedreader is not enabled, so show the bubble about enabling
-        //       the feature globally.
-        //   (2) Speedreader is enabled and the domain is blacklisted. Show the
-        //       Speedreader bubble to let the user remove from the list.
-        //   (3) Speedreader is enabled and the domain doesn't match the url
-        //       heuristic. Don't show any bubble.
-        if (speedreader_enabled) {
-          if (!tab_helper->IsEnabledForSite())
-            tab_helper->ShowSpeedreaderBubble();
-        } else {
-          tab_helper->ShowReaderModeBubble();
-        }
-      } else {
-        // The user clicked the reader mode icon to distill a page. If
-        // Speedreader is not enabled then automatically show the bubble
-        // prompting the user to turn on the feature.
-        //
-        // TODO(keur): Maybe register a pref to only automatically drop the
-        // reader mode bubble once. The user can always get to it manually
-        // later.
+        break;
+      case DistillState::kReaderMode:
+        tab_helper->ShowReaderModeBubble();
+        break;
+      case DistillState::kPageProbablyReadable: {
+        auto* speedreader_service =
+            speedreader::SpeedreaderServiceFactory::GetForProfile(
+                browser->profile());
         tab_helper->SingleShotSpeedreader();
-        if (!speedreader_enabled)
+        if (speedreader_service->ShouldPromptUserToEnable()) {
           tab_helper->ShowReaderModeBubble();
+          speedreader_service->IncrementPromptCount();
```

Comment:
I was thinking that maybe we can move this logic from `browser_commands.cc`? seems too complex to sit there

Reply:
fixed